grad_login/lib/infrastructure/medicine/interactions_service.dart

```diff
@@ -23,7 +24,16 @@ class InteractionsService {
         "Authorization": "JWT $token",
       },
     );
-    final responseData = json.decode(response.body);
+    if (response.statusCode == 500) {
+      responseData['error'] =
```

Comment:
You should not mutate the response data at any time, can you split this into another variable? or another object?